emscripten.py

```diff
@@ -1907,31 +1907,34 @@ def debug_copy(src, dst):
         cmd += ['--export', export[1:]] # Strip the leading underscore
     shared.check_call(cmd)
 
-    if settings['DEBUG_LEVEL'] >= 2 or settings['PROFILING_FUNCS']:
-      if settings['DEBUG_LEVEL'] >= 4:
-        sourcemap_cmd = [shared.PYTHON, path_from_root('tools', 'wasm-sourcemap.py'), 
-                         base_wasm,
-                         '--dwarfdump=' + shared.LLVM_DWARFDUMP,
-                         '-o',  base_wasm + '.map']
-        if not settings['SOURCE_MAP_BASE']:
-          logging.warn("Wasm source map won't be usable in a browser without --source-map-base")
-        else:
-          sourcemap_cmd += ['--source-map-url=' + settings['SOURCE_MAP_BASE'] + os.path.basename(settings['WASM_BINARY_FILE']) + '.map']
-        shared.check_call(sourcemap_cmd)
-
     if DEBUG:
       logging.debug('  emscript: lld took %s seconds' % (time.time() - t))
       t = time.time()
     debug_copy(base_wasm, 'base_wasm.wasm')
 
+    if settings['DEBUG_LEVEL'] >= 4:
+      source_map = base_wasm + '.map'
+      sourcemap_cmd = [shared.PYTHON, path_from_root('tools', 'wasm-sourcemap.py'), 
+                       base_wasm,
+                       '--dwarfdump=' + shared.LLVM_DWARFDUMP,
+                       '-o',  source_map]
+      if not settings['SOURCE_MAP_BASE']:
+        logging.warn("Wasm source map won't be usable in a browser without --source-map-base")
+      else:
+        sourcemap_cmd += ['--source-map-url=' + settings['SOURCE_MAP_BASE'] + os.path.basename(settings['WASM_BINARY_FILE']) + '.map']
+      shared.check_call(sourcemap_cmd)
+      debug_copy(source_map, 'base_wasm.map')
+
     cmd = [wasm_emscripten_finalize, base_wasm, '-o', wasm,
            '--global-base=%s' % shared.Settings.GLOBAL_BASE,
            ('--emscripten-reserved-function-pointers=%d' %
             shared.Settings.RESERVED_FUNCTION_POINTERS)]
-    if settings['DEBUG_LEVEL'] >= 2 or settings['PROFILING_FUNCS']:
-      cmd += ['-g', '--input-source-map=' + base_wasm + '.map']
-      cmd += ['--output-source-map=' + wasm + '.map']
+    if settings['DEBUG_LEVEL'] >= 4:
```

Comment:
Hm, all this repetition of `if settings['DEBUG_LEVEL'] >= 4` makes me think we should hold that in a named variable, `should_use_source_maps` or something

Reply:
Maybe.. something in settings perhaps.  Maybe thats a sepeate change though.

Reply:
Oh I mean just local to this function, like
```python
should_use_source_maps = settings['DEBUG_LEVEL'] >= 4
if should_use_source_maps:
  # yada yada
```